cloudflare_rules/cloudflare_firewall_high_volume_events_greynoise.py

```diff
@@ -0,0 +1,42 @@
+from ipaddress import ip_address
+
+from panther_cloudflare_helpers import map_source_to_name
+from panther_greynoise_helpers import GetGreyNoiseObject, GetGreyNoiseRiotObject
+
+
+def rule(event):
+    if event.get("Action") != "block":
+        return False
+
+    # Validate the IP is actually an IP
+    try:
+        ip_address(event.get("ClientIP"))
+    except ValueError:
+        return False
+
+    # If IP is in the RIOT dataset, we can assume safe
+    global NOISE  # pylint: disable=global-variable-undefined
+    NOISE = GetGreyNoiseObject(event)
+    riot = GetGreyNoiseRiotObject(event)
+    if riot.is_riot("ClientIP"):
+        return False
+
+    # Check if IP is classified as 'malicious'
+    if NOISE.classification("ClientIP") == "malicious":
+        return True
+    return False
+
+
+def title(event):
+    return (
+        f"High Volume Events Detected - "
```

Comment:
Sorry, I missed this in my first review. I think adding in the word "Malicious" to the alert title, rule name and filename would be more descriptive.

Reply:
This is similar to the `cloudflare_rules/cloudflare_firewall_suspicious_event_greynoise.py` detection, the main difference being on whether or not the event was blocked. I specifically avoided `malicious` and `suspicious` for this detection as the events are blocked